feat: 구매자는 REST API를 통해서 포인트 충전 요청을 할 수 있다.

yudonggeun opened the pull request:
## 📄 Summary
### 변경사항
- [구매자 포인트 충전 API 구현](https://github.com/woowa-techcamp-2024/Team7-ELEVEN/commit/86fad39f2ccbd608cf4c97a3f8d873112b16205a)

### 테스트
- [구매자 충전 API 성공 케이스 테스트 작성](https://github.com/woowa-techcamp-2024/Team7-ELEVEN/commit/203af244b51c6f57a2ff2c34b7a076b6d7205396)
- [DocumentationTest 설정](https://github.com/woowa-techcamp-2024/Team7-ELEVEN/commit/5e78975457c009bae88153516534703b7682e133)
     - `MockMvc`를 적용한 이유는 세션을 등록하는 과정에서 Object의 타입이 보존이 되지 않고 String으로 변환되어서 테스트가 진행이 되는 예외 케이스 때문에 사용했습니다.
- [MemberFixture 빌더 패턴 적용](https://github.com/woowa-techcamp-2024/Team7-ELEVEN/commit/17b366178ac2c23188f816dbaf7c8ca3d61ed0e3)

## 🙋🏻 More

>


close #155

yudonggeun committed:
test: MemberFixture 빌더 패턴 적용

- 동적인 Fixture 생성을 위한 빌더 패턴을 적용합니다.

yudonggeun committed:
feat: 구매자 포인트 충전 요청 request body dto 생성

yudonggeun committed:
test: DocumentationTest 설정

- PaymentService 필드 추가
- PaymentController 등록
- MockMvc 적용

yudonggeun committed:
feat: PaymentService 빈 등록

yudonggeun committed:
feat: 구매자 포인트 충전 API 구현

yudonggeun committed:
test: 구매자 충전 API 성공 케이스 테스트 작성

HiiWee commented on src/test/java/com/wootecam/luckyvickyauction/core/payment/controller/PaymentControllerTest.java:
```diff
@@ -0,0 +1,45 @@
+                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                        .accept(MediaType.APPLICATION_JSON_VALUE)
+                        .cookie(new Cookie("JSESSIONID", "sessionId"))
+                        .sessionAttr("member", member)
```
session key는 signInMember로 되어 있습니다!

HiiWee commented on src/test/java/com/wootecam/luckyvickyauction/core/payment/controller/PaymentControllerTest.java:
```diff
@@ -0,0 +1,45 @@
+                        requestCookies(
+                                cookieWithName("JSESSIONID").description("세션 ID")
+                        ),
+                        requestFields(
+                                fieldWithPath("amount").description("충전할 포인트 금액")
+                        )
+                ));
```
요기다가 amount에 대한 타입 명시해주면 좋을것 같아요!

HiiWee reviewed (commented):
굳굳!! 의견 몇개 남겼어요 확인 부탁드립니다!

yudonggeun committed:
Merge branch 'refs/heads/dev' into feat/155-issue

# Conflicts:
#	src/test/java/com/wootecam/luckyvickyauction/documentation/DocumentationTest.java

yudonggeun committed:
test: pr 리뷰 반영
- https://github.com/woowa-techcamp-2024/Team7-ELEVEN/pull/222#pullrequestreview-2244230206

HiiWee commented on src/test/java/com/wootecam/luckyvickyauction/core/payment/controller/PaymentControllerTest.java:
```diff
@@ -38,7 +38,7 @@ class PaymentControllerTest extends DocumentationTest {
                                 cookieWithName("JSESSIONID").description("세션 ID")
                         ),
                         requestFields(
-                                fieldWithPath("amount").description("충전할 포인트 금액")
+                                fieldWithPath("amount").type("Number").description("충전할 포인트 금액")
                         )
```
`JsonFieldType.NUMBER` 이라는 Enum을 제공합니다!!

yudonggeun committed:
test: JsonFieldType 적용

yudonggeun committed:
Merge branch 'dev' into feat/155-issue

HiiWee committed:
Merge branch 'dev' into feat/155-issue

HiiWee reviewed (dismissed):
LGTM!! 고생하셨습니다 ~~!

yudonggeun committed:
Merge branch 'refs/heads/dev' into feat/155-issue

yudonggeun committed:
fix: 머지 이후 테스트 실패로 발견한 버그 수정

minseok-oh reviewed (approved):
LGTM!! 👍
